### ISSUE TITLE
Bidirectional LTSM example throws shape error

### ISSUE THREAD

**Issue by colinskow** [tflearn/tflearn#818]
Running the following file throws an error:
https://github.com/tflearn/tflearn/blob/master/examples/nlp/bidirectional_lstm.py

`ValueError: Shape (128, ?) must have rank at least 3`

Setup:
- MacOS Sierra (10.12)
- Python 2.7
- Tensorflow v1.2.0
- TFLearn v0.3.2

```
Traceback (most recent call last):
  File "bidirectional_lstm.py", line 47, in <module>
    net = bidirectional_rnn(net, BasicLSTMCell(128), BasicLSTMCell(128))
  File "/Users/colin/tensorflow/lib/python2.7/site-packages/tflearn/layers/recurrent.py", line 374, in bidirectional_rnn
    dtype=tf.float32)
  File "/Users/colin/tensorflow/lib/python2.7/site-packages/tensorflow/python/ops/rnn.py", line 375, in bidirectional_dynamic_rnn
    time_major=time_major, scope=fw_scope)
  File "/Users/colin/tensorflow/lib/python2.7/site-packages/tensorflow/python/ops/rnn.py", line 574, in dynamic_rnn
    dtype=dtype)
  File "/Users/colin/tensorflow/lib/python2.7/site-packages/tensorflow/python/ops/rnn.py", line 637, in _dynamic_rnn_loop
    for input_ in flat_input)
  File "/Users/colin/tensorflow/lib/python2.7/site-packages/tensorflow/python/ops/rnn.py", line 637, in <genexpr>
    for input_ in flat_input)
  File "/Users/colin/tensorflow/lib/python2.7/site-packages/tensorflow/python/framework/tensor_shape.py", line 649, in with_rank_at_least
    raise ValueError("Shape %s must have rank at least %d" % (self, rank))
ValueError: Shape (128, ?) must have rank at least 3
```

**Comment by willduan** [comment]
Please try to install tensorflow 1.0 or 1.1.  Since it is not very compatible with tensorflow 1.2 version currently.

**Comment by colinskow** [comment]
I can confirm this does work fine with TensorFlow 1.1.0. That means there was a breaking change in `tensorflow.python.ops.rnn.bidirectional_dynamic_rnn`. I would like to get this working in the latest version since I am relying on new 1.2.0 features.

**Comment by colinskow** [comment]
Breaking changes to RNNCell are documented here:
https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md#release-120

**Comment by colinskow** [comment]
I found the problem:
On TF <= 1.1.0 `recurrent.py` imports `static_bidirectional_rnn` which works.
On TF >= 1.2.0 `recurrent.py` imports `bidirectional_dynamic_rnn` which is not compatible with the current implementation.

Removing the input for `bidirectional_dynamic_rnn` solves it. PR inbound.

**Comment by stepsma** [comment]
Hi @colinskow 

Could you please re-visit this problem one more time, I am currently still getting this error with tensorflow 1.4. Other people seems seeing this problem as well with more recent tensorflow version (https://github.com/tflearn/tflearn/issues/988). Thank you for your concern.

Best,

**Comment by FTAsr** [comment]
I'm getting the same error using the following packages on a linux GPU machine, a conda environment with python 2.7 (same code gives no error on my MacBook):

tensorflow                1.3.0                         0  
tensorflow-base           1.3.0            py27h0dbb4d0_1  
tensorflow-gpu            1.3.0                         0  
tensorflow-gpu-base       1.3.0           py27cuda8.0cudnn6.0_1  
tensorflow-tensorboard    0.1.5                    py27_0  
tflearn                   0.3.2                     <pip>

**Comment by MrKZZ** [comment]
Oh, I have this problem too! Seems that this problem hasn't fixed so far.Neither tensorflow or tflearn team wants to do this job! But I wander to know what's the problem means?
ValueError: Shape (128, ?) must have rank at least 3

**Comment by Johy** [comment]
Same problem here, some insight would be nice.

**Comment by thomas-young-2013** [comment]
same problem found!  I am working with tensorflow_gpu 1.8.0, please revisit the issue again!

**Comment by maxiaomu** [comment]
Oh, i've found the same problem with tf.__version__
'1.6.0-rc0'.

**Comment by greg234234** [comment]
aye, here too... tensorflow 1.8.0

**Comment by sandove** [comment]
same error... tensorflow 1.9.0

**Comment by dyq0811** [comment]
same error here!...

**Comment by zhvankh** [comment]
https://github.com/tflearn/tflearn/commit/d01a0b9f59eb8da2e5db9f5fc64617c6379d1ab2#diff-5885c0e8432a40cf7a3e4b2448302e3a
this is best answer for this problem

**Comment by Vani261196** [comment]
same error here ....... tensorflow 1.14

**Comment by Syauri** [comment]
Getting the same error in Tensorflow 1.15! Please revisit!

**Comment by yunfanye** [comment]
FWIW, I met this issue today too and it turned out it was a different root cause that led to the same error message. Given that there was a comment saying it was fixed in TF >= 1.2, I'd bet that the issue was indeed fixed and all the later errors were caused by other issues.

It was so frustrating initially to see all the comments here complaining a issue 5 years ago without answers, so I figured I'd shed some light.

The error message basically says: the code expects a 3-dimensional input, i.e. (128, ?, ?) instead of (128, ?). For my specific error, it was caused by an upstream bug where one dimension was gone, i.e. it should have been (128, 256, ?).
